server/src/main/java/org/opensearch/action/search/SearchRequestSlowLog.java

```diff
@@ -108,12 +109,18 @@ public final class SearchRequestSlowLog extends SearchRequestOperationsListener
 
     private static final ToXContent.Params FORMAT_PARAMS = new ToXContent.MapParams(Collections.singletonMap("pretty", "false"));
 
-    public SearchRequestSlowLog(ClusterService clusterService) {
-        this(clusterService, LogManager.getLogger(CLUSTER_SEARCH_REQUEST_SLOWLOG_PREFIX)); // logger configured in log4j2.properties
+    @Inject
```

Comment:
You're explicitly passing the SearchRequestListenerManager from the `Node` constructor so you don't need `Inject`.

In fact, why don't you just pass all the "standard" listeners to the `SearchRequestListenerManager`'s constructor? 

Then plugins can get a reference to `SearchRequestListenerManager` so that their `RequestListenerFactory` can register any listeners? (Or the `Node` constructor can ask all `RequestListenerPlugin`s for their `RequestListeners` and it can call `addListener` on the manager.)

Reply:
Yes it would be much easier if we explicitly add all standard listeners to the SearchRequestListenerManager and set enable when necessary. I have just pushed a new revision for that.